app/models/url.rb

```diff
@@ -2,9 +2,10 @@ class Url < ActiveRecord::Base
   scope :popular, -> { order(popularity: :desc) }
   scope :newest, -> { order(:created_at) }
 
-  before_save { |url| url.short = url.generate_short_url }
+  before_save :generate_short_url
 
   def generate_short_url
-    ('a'..'z').to_a.shuffle[0,5].join
+    url.short = ('a'..'z').to_a.shuffle[0,5].join
```

Comment:
Space missing after comma.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.